File: src/main/prim/_list.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim

import org.nlogo.core.Syntax
import org.nlogo.api.{ LogoException, LogoList, LogoListBuilder }
import org.nlogo.nvm.{ Reporter, Context, Pure, CustomGenerated }

class _list extends Reporter with Pure with CustomGenerated {
  override def syntax =
    Syntax.reporterSyntax(
      right = List(Syntax.RepeatableType | Syntax.WildcardType),
      ret = Syntax.ListType,
      defaultOption = Some(2),
      minimumOption = Some(0))
  override def report(context: Context): LogoList = {
    val builder = new LogoListBuilder
    var i = 0
    while(i < args.length) {
      builder.add(args(i).report(context))
      i += 1
    }
    builder.toLogoList
  }
}

File: netlogo-gui/src/main/prim/_list.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim;

import org.nlogo.api.LogoException;
import org.nlogo.api.LogoListBuilder;
import org.nlogo.core.Syntax;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;

public final strictfp class _list
    extends Reporter
    implements org.nlogo.core.Pure, org.nlogo.nvm.CustomGenerated {

  @Override
  public int returnType() {
    return Syntax.ListType();
  }



  @Override
  public Object report(final Context context) throws LogoException {
    LogoListBuilder list = new LogoListBuilder();
    for (int i = 0; i < args.length; i++) {
      list.add(args[i].report(context));
    }
    return list.toLogoList();
  }
}
